harness/determined/layers/_worker_process.py

```diff
@@ -231,14 +245,31 @@ def _launch_python_subprocess(self) -> subprocess.Popen:
         return subprocess.Popen(python_cmd)
 
     def _do_startup_message_sequence(self) -> None:
-        # Wait for a ReadyMessage from every worker.
+        # Wait for a ConnectedMessage from every worker.
         responses, exception_received = self.broadcast_server.gather_with_polling(
             self._health_check
         )
 
         if exception_received:
             raise det.errors.WorkerError("Training process died.")
 
+        for response in responses:
+            check.is_instance(
+                response,
+                ipc.ConnectedMessage,
+                f"Did not receive ConnectedMessage from worker. Got: {response}",
+            )
+            response = cast(ipc.ConnectedMessage, response)
+            self._worker_process_ids.append(response.process_id)
+
+        # Send an Ack so that worker process is able to send ReadyMessage once it is ready.
+        self.broadcast_server.broadcast(ipc.ConnectedMessageAck())
```

Comment:
Yeah, I'm pretty sure everything from these lines to the end of this function could just go away...

Reply:
yep went away